common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3226,6 +3171,71 @@ object CelebornConf extends Logging {
 
+  val SPARK_SHUFFLE_WRITER_MODE: ConfigEntry[String] =
+    buildConf("celeborn.client.spark.shuffle.writer")
+      .withAlternative("celeborn.shuffle.writer")
+      .withAlternative("rss.shuffle.writer.mode")
+      .categories("client")
+      .doc("Celeborn supports the following kind of shuffle writers. 1. hash: hash-based shuffle writer " +
+        "works fine when shuffle partition count is normal; 2. sort: sort-based shuffle writer works fine " +
+        "when memory pressure is high or shuffle partition count is huge.")
+      .version("0.3.0")
+      .stringConf
+      .transform(_.toUpperCase(Locale.ROOT))
+      .checkValues(Set(ShuffleMode.HASH.name, ShuffleMode.SORT.name))
+      .createWithDefault(ShuffleMode.HASH.name)
+
+
+  val CLIENT_PUSH_UNSAFEROW_FASTWRITE_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.push.unsafeRow.fastWrite.enabled")
+      .categories("client")
+      .version("0.2.2")
+      .doc("This is Celeborn's optimization on UnsafeRow for Spark and it's true by default. " +
+        "If you have changed UnsafeRow's memory layout set this to false.")
+      .booleanConf
+      .createWithDefault(true)
+
+  val SPARK_SHUFFLE_FORCE_FALLBACK_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.shuffle.forceFallback.enabled")
+      .withAlternative("celeborn.shuffle.forceFallback.enabled")
+      .withAlternative("rss.force.fallback")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Whether force fallback shuffle to Spark's default.")
+      .booleanConf
+      .createWithDefault(false)
+
+  val SPARK_SHUFFLE_FORCE_FALLBACK_PARTITION_THRESHOLD: ConfigEntry[Long] =
+    buildConf("celeborn.client.spark.shuffle.forceFallback.numPartitionsThreshold")
+      .withAlternative("celeborn.shuffle.forceFallback.numPartitionsThreshold")
+      .withAlternative("rss.max.partition.number")
+      .categories("client")
+      .version("0.3.0")
+      .doc(
+        "Celeborn will only accept shuffle of partition number lower than this configuration value.")
+      .longConf
+      .createWithDefault(500000)
+
+  val CLIENT_PUSH_SORT_MEMORY_THRESHOLD: ConfigEntry[Long] =
+    buildConf("celeborn.client.spark.push.sort.memory.threshold")
+      .withAlternative("celeborn.push.sortMemory.threshold")
+      .withAlternative("rss.sort.push.data.threshold")
+      .categories("client")
+      .doc("When SortBasedPusher use memory over the threshold, will trigger push data.")
+      .version("0.3.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefaultString("64m")
+
+  val CLIENT_PUSH_SORT_PIPELINE_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.push.sort.pipeline.enabled")
+      .withAlternative("celeborn.push.sort.pipeline.enabled")
+      .categories("client")
+      .doc("Whether to enable pipelining for sort based shuffle writer. If true, double buffering" +
+        " will be used to pipeline push")
+      .version("0.3.1")
```

Comment:
0.3.0 is not released yet, why 0.3.1?

Reply:
is flink possible to have such optimization

Reply:
> is flink possible to have such optimization

No

Reply:
> 0.3.0 is not released yet, why 0.3.1?

Strange...seems someone make a mistake before